### DISCUSSION TITLE
Amalgamated vs parallel build time.

### DISCUSSION THREAD

**Discussion by zevv** [janet-lang/janet#1173] [General]
When working on the Janet source, I often find my edit-to-run turnaround time just a bit too high to be comfy; even though I have a fairly modern 24 core machine, the final build of the amalgamated `janet.c` still takes up to 6 seconds because this last step can not be parallelized - shame!

As a workaround I have some local changes to my `Makefile` and `boot.janet`  (see below) that adds a target for generating the boot code without amalgamating all the sources, and just building and linking all the individual objects the "old fashioned" way. This results in a build time of approx 1.5 seconds, instead of the 8 seconds of the normal target.

Are other people also bothered by the compilation time, and would it make sense to add a parallelizable build target to the Makefile for this?

https://github.com/zevv/janet/commit/8f43bf31d7f835355cf355665b22dd3f551f23af

**Comment by sogaiu** [comment]
Nice speedup!

Perhaps it's because my machine isn't as nice (or may be I didn't do it right), but my speed up wasn't so dramatic.  Somewhere in the 1-2 seconds range.

Oh wait...no, it doesn't look faster here.  Must be doing something wrong.

**Reply by zevv** [reply]
You're doing a parallel build, right? try (replace `NCPUS` with a sane value for your machine)
```
make clean; time make -j NCPUS
``` 
vs
```
make clean; time make -j NCPUS quicky=1
```

**Reply by sogaiu** [reply]
Oh yes, pilot error indeed.

I get the same sort of numbers you reported...nice :)

Thanks!
